tools/pylib/boutdata/data.py

```diff
@@ -141,6 +144,58 @@ def __str__(self, indent=""):
             text += indent + " |- " + self._sections[s].__str__(indent+" |  ")
         return text
 
+    def evaluate_scalar(self, name):
+        """
+        Evaluate (recursively) scalar expressions
+        """
+        expression = self._substitute_expressions(name)
+
+        # replace ^ with ** so that Python evaluates exponentiation
+        expression = expression.replace("^","**")
+
+        # use numpy functions
+        for func in funcstoreplace:
+            expression = re.sub(r"\b"+func.lower()+r"\b", "numpy."+func, expression)
```

Comment:
Could these just be `import`ed from `numpy`? That would remove these two lines, plus `funcstoreplace`